src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py

```diff
@@ -6,64 +6,196 @@
 #
 import random as _random
 import turicreate as _tc
+import turicreate.toolkits._internal_utils as _tkutl
 from turicreate import extensions as _extensions
 from turicreate.toolkits._model import CustomModel as _CustomModel
+from turicreate.toolkits._model import PythonProxy as _PythonProxy
+from turicreate.toolkits.object_detector.object_detector import ObjectDetector as _ObjectDetector
 
 def create(dataset,
            target,
+           augmentation_mode="auto",
+           flip_horizontal=True,
+           flip_vertical=True,
            backgrounds=None,
-           feature=None,
            batch_size=0,
            max_iterations=0,
            seed=None,
-           verbose=True):
-    model = _extensions.one_shot_object_detector()
+           verbose=True,
+           **kwargs):
+    """
+    Create a :class:`ObjectDetector` model.
+
+    Parameters
+    ----------
+
+    dataset : SFrame | tc.Image
+        An SFrame that contains all the starter images along with their
+        corresponding labels.
+        If the dataset is a single tc.Image, this parameter is just that image.
+        Every starter image should entirely be just the starter image without
+        any padding.
+        RGB and RGBA images allowed.
+
+    target : string
+        The target column name in the SFrame that contains all the labels. 
+        If the dataset is a single tc.Image, this parameter is just a label for
+        that image.
+
+    augmentation_mode : string optional
+        An augmentation mode is a shortcut to setting the 'flip_horizontal',
+        'flip_vertical', and other advanced parameters in **kwargs.
+        For example, an augmentation_mode of 'auto' automatically sets the
+        following parameter values:
+            flip_horizontal = True
+            flip_vertical = True
+            yaw = [(0, 360)]
+            pitch = [(0, 360)]
+            roll = [(0, 360)]
+            blur = True
+            noise = True
+            lighting_perturbation = True
+        Here is the set of augmentation modes available:
+        - 'auto'
+        - 'logo'
+        - 'road_sign'
+        - 'playing_card'
+        - 'chart'
+        TBD/TODO: Specify the different pre-set parameter values each of these
+        modes corresponds to, probably as a table.
+        If 'dataset' is an SFrame, the augmentation_mode specified as this
+        parameter will be applied to all the images provided in the SFrame.
+        If you'd like different images in the 'dataset' SFrame to have different
+        augmentation_mode's, those modes can be specified as a column in the
+        SFrame, and that column name should be passed in to the
+        'augmentation_mode' parameter.
+
+    flip_horizontal : bool optional
+        Boolean flag that indicates whether the object can be flipped
+        horizontally,
+        i.e. whether a water image of the object should appear can be in the
```

Comment:
What does "water image" mean?

Note that the OD toolkit has an advanced parameter `aug_horizontal_flip`, which controls whether data augmentation can reflect the image horizontally, across the vertical line passing through the image's center. Are you giving this parameter the opposite meaning?

Reply:
I meant `flip_horizontal` to mean whether data augmentation can reflect the image vertically, along the horizontal line passing through the image's center. In other words, whether the user would like to see a _"water image" (reflection in water)_  of the image passed in, in the augmented data.

         o 
          |
         / \
----------------- water
         \ /
          |
         o